### ISSUE TITLE
Support for alternative repository hosts and registries

### ISSUE THREAD

**Issue by steveoh** [greenkeeperio/greenkeeper#92]
Edit by @boennemann: To avoid an ever repeating discussion please use GitHub's reaction feature to express your support for a certain service.
- ❤️ -> +1 for Bower
- 😄 -> +1 for Gitlab
- 🎉 -> +1 for Bitbucket

---

Can we keep a list of planned support repositories?
- Bower
- Gitlab #86
- BitBucket #90
- Etc?

**Comment by boennemann** [comment]
Thank you for your feedback. While possible we can't yet afford to work on all of these services, or don't think that they would make sense for Greenkeeper. Here is a (not necessarily complete) list of services and their state:

## Working

GitHub

## Work in Progress

GitHub Enterprise

## Future/Maybe

GitLab
Bitbucket
every other hosted git service with pull requests

## Probably not

Bower
Git dependencies

**Comment by klarkc** [comment]
Hello @boennemann,

Why not support bower packages, there is something blocking this besides the work needed to do this?

**Comment by janl** [comment]
@klarkc the *significant* work needed to do this.

besides, bower has no central registry, so we don’t know about package updates as easily as from npm, so we’d have to figure out something for that, like integrate with libraries.io, which adds another dependency on our service etc. etc. — so yeah, no Bower any time soon, sorry!

**Comment by firrae** [comment]
I would really appreciate GitLab support. With them now offering free CI services it would make it quite the complete package.

**Comment by boennemann** [comment]
@firrae Thanks for your feedback here. As I said we can't yet afford to work on all of these services, but we might in future. We know that this would be really great for some of you, but we don't want to make any false promises.

**Comment by janl** [comment]
To expand on @boennemann’s answer: we have to make a living too, and putting a lot of work into supporting a server that only gets us more free users is not something we can commit to right now.

We are a small company, have no venture capital (and don’t plan on raising any), so we are only accountable to our users and customers and not going to get sold to a big company in a year or two and then shut down or “sunsetted”.

The flip side of this coin is that development is slower than you might be used to from other services that have taken the above road. We hope you understand!

Eventually, we’d love to be able to support them all, of course :)

**Comment by firrae** [comment]
@boennemann @janl haha, that's cool duders, I understand that, I'm in the same boat myself. Just figured I'd +1 that for the future. Keep it up, tool works great even with my weird setup via Meteor!

**Comment by unional** [comment]
Supporting private bitbucket would be a good revenue generator. :stuck_out_tongue:

**Comment by shamasis** [comment]
I will be a customer right away with Bitbucket in place. 😜

**Comment by MrHayato** [comment]
Can support for custom npm registries also be added? We host our own private npm registry and would love to use greenkeeper against it

**Comment by shamasis** [comment]
Um closed without any explanation? 😞

**Comment by steveoh** [comment]
I was cleaning house on all of my open issues.

**Comment by shamasis** [comment]
We have a bitbucket and private NPM stack and would love to use greenkeeper with it.

**Comment by travi** [comment]
for those following this thread, private npm registries do work today as long as the published package calls the `greenkeeper postpublish` script. my team publishes private packages to nexus successfully. the current limitation is that the code repository webhooks only work with github.

**Comment by incubus8** [comment]
any status on this? it's been a year..

**Comment by abalad** [comment]
I found a solution that seems to have the same proposition. Note: I have not tested it yet. https://requires.io/

**Comment by jakubzitny** [comment]
@abalad that one seems to be python-only

**Comment by iamstarkov** [comment]
hey, @boennemann can my company contribute somehow to get support for bitbucket? (money, time, developers)

**Comment by KillaPL** [comment]
My company would like to contribute developer time for Gitlab integration. How would we go about this?

**Comment by gr2m** [comment]
Thanks Łukasz, we will get back to you 👍

**Comment by KillaPL** [comment]
We were informed about https://github.com/singapore/renovate, currently trying to get that working.

**Comment by iamstarkov** [comment]
@gr2m can my company help you as well?

**Comment by gr2m** [comment]
@iamstarkov absolutely :) stay tuned

**Comment by kellyrmilligan** [comment]
🎉

**Comment by iamstarkov** [comment]
@gr2m why did you delete @KillaPL's comment about https://github.com/singapore/renovate?

<img width="751" alt="screen shot 2017-11-17 at 9 53 46 pm" src="https://user-images.githubusercontent.com/559321/32968564-000cc0be-cbe2-11e7-9206-c8f9118cc637.png">

vs

<img width="876" alt="screen shot 2017-11-17 at 9 54 59 pm" src="https://user-images.githubusercontent.com/559321/32968571-081e188e-cbe2-11e7-9aa3-f21364142e52.png">

**Comment by iamstarkov** [comment]
and yes, in absence of greenkeeper's response for features, i and my company also will try https://github.com/singapore/renovate

**Comment by gr2m** [comment]
I’ve reached out to Łukasz explaining why immediately after I deleted the comment. I’m sorry for the lack of transparency, we don’t yet have the proper process in place. It looked like shady marketing, so I decided to remove it, I should have asked to self-moderate instead. My apologies. 

I would appreciate if you would moderate your comments by putting them into `<details>` tag, as the comments are off topic. I’ll do the same with mine, to help keep the discussion on topic. The way it works is that you wrape your comments in a <details> tag like this

```
<details>
  <summary>
  off topic
  </summary>
  Original comment here
</details>
```

Thanks :)
